src/lib.rs

```diff
@@ -107,3 +113,46 @@ unsafe fn secbuf_desc(bufs: &mut [sspi::SecBuffer]) -> sspi::SecBufferDesc {
         pBuffers: bufs.as_mut_ptr(),
     }
 }
+
+unsafe fn alpn_list(protos: &[Vec<u8>]) -> Vec<u8> {
+    // ALPN wire format is each ALPN preceded by its length as a byte.
+    let mut alpn_wire_format = Vec::with_capacity(
+        protos.iter().map(Vec::len).sum::<usize>() + protos.len(),
+    );
+    for alpn in protos {
+        alpn_wire_format.push(alpn.len() as u8);
+        alpn_wire_format.extend(alpn);
+    }
+
+    // Make sure that the memory we're using for `sspi::SEC_APPLICATION_PROTOCOLS` matches
```

Comment:
`SEC_APPLICATION_PROTOCOLS` technically has a 4-byte-alignment requirement, but given that we're manually aligning the memory here, I figured I would make it 8 anyhow in case it changes in the future and we forget to update this.

I can make it 4 bytes, or add a comment.

Reply:
I would use the compiler where possible directly, e.g. `mem::align_of::<SEC_APPLICATION_PROTOCOLS>()` directly.
See below, Best I come up with is either using sth. like the following:
```
let buf: Vec<u8> = Vec::with_capacity(mem::align_of::<SEC_APPLICATION_PROTOCOLS>() + size);
```
and then using the slice `&buf[ptr::align_offset(mem::align_of::<SEC_APPLICATION_PROTOCOLS>())..]` everywhere.
Alterantively `std::alloc` with a `SEC_APPLICATION_PROTOCOLS` wrapper that deals with custom alignment allocation/deallocation.